lib/plugins/filter/template_locals/i18n.ts

```diff
@@ -13,9 +13,9 @@ function i18nLocalsFilter(this: Hexo, locals: LocalsType): void {
 
   if (!lang) {
     const pattern = new Pattern(`${i18nDir}/*path`);
-    const data = pattern.match(locals.path);
+    const data = pattern.match(locals.path) as Record<string, any>;
 
-    if (data && data.lang && i18nLanguages.includes(data.lang)) {
+    if (data && typeof data.lang === 'string' && i18nLanguages.includes(data.lang)) {
```

Comment:
Use `"lang" in data` to properly guard the type, not use `as Record`

Reply:
when i remove `as Record<string, any>;` from `data` variable got this error

![image](https://github.com/user-attachments/assets/cad53802-ace5-435b-916d-2f77bbc334ab)

Reply:
when i cast it to `any`, no error showing and successful build.

![image](https://github.com/user-attachments/assets/38d9e34e-fe3e-4a3a-a8b9-b40a0abc8df4)

Reply:
Do `typeof data === 'object' && 'lang' in data`. You can always narrow the type down. Do not cheat using `as` or `any`.

Reply:
> Do `typeof data === 'object' && 'lang' in data` then. You can always narrow the type down. Do not cheat using `as` or `any`.

`typeof data === 'object'` it makes property of object `unknown`

![image](https://github.com/user-attachments/assets/c06a290b-f811-4dc5-98bf-8e3cf03a3af6)

Reply:
> Do `typeof data === 'object' && 'lang' in data` then. You can always narrow the type down. Do not cheat using `as` or `any`.

`RegExpMatchArray` also result `object` when using `typeof`, also for Array, and null when you check using `typeof` it return `object`. This is typescript

Reply:
You cannot generalizing type variable checking by `object`.
For example `Buffer` when you check it using `typeof` the result also `object`. You should using `Buffer.isBuffer` to check the variable is buffer or not.
And array, you should using `Array.isArray` to check that is type of array or not.
when you generalized all to `typeof variable === object` it will always true.
That why, typescript makes data to error. because of this, and the `data` variable has 3 different types including Array, object and boolean.

Reply:
i dont know again to keep this working without enforcing cast to `any`, this caused by tsconfig.json `strict` rule.

![image](https://github.com/user-attachments/assets/8c7773fb-c541-45e8-afb0-be9c9d10e494)

fail to run `npm run build` when not enforcing cast to any